planner/core/common_plans.go

```diff
@@ -496,6 +496,14 @@ REBUILD:
 	e.names = names
 	e.Plan = p
 	_, isTableDual := p.(*PhysicalTableDual)
+	if !isTableDual {
+		// Some MaybeOverOptimized push-down filter conditions will be copied and kept in the upper layer.
+		// So the plan for 'Sel -> TableDual' should not be cached either
+		sel, isSelection := p.(*PhysicalSelection)
+		if isSelection {
+			_, isTableDual = sel.children[0].(*PhysicalTableDual)
+		}
+	}
```

Comment:
How about creating a new function like `hasTableDual` to check whether a plan has `TableDual`?